expression/builtin_convert_charset.go

```diff
@@ -65,6 +66,10 @@ func (c *tidbToBinaryFunctionClass) getFunction(ctx sessionctx.Context, args []E
 			return nil, err
 		}
 		bf.tp = args[0].GetType().Clone()
+		// adjust enum/set type
+		if bf.tp.Tp == mysql.TypeEnum || bf.tp.Tp == mysql.TypeSet {
+			bf.tp.Tp = mysql.TypeString
+		}
```

Comment:
```suggestion
		bf.tp.Tp = mysql.TypeVarString
```

`to_binary` should always return string type.